runtime/criusupport/criusupport.c

```diff
@@ -237,46 +231,70 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 			log = J9_JNI_UNWRAP_REFERENCE(logFile);
```

Comment:
Move this block of code that unwraps the `jstring` to the point where it will be used.  Best not to have naked object pointers across calls to helpers.  It's easier to reason about the code if jni handles are only unwrapped at point of use

Reply:
Or in this case, it might be better to do all the `getNativeString` calls up front so you don't have to worry about the objects later in this code.  It will also clean up the logic as all conversion is done in a block and no directories are opened unless all conversions succeed.

Reply:
Moved the calls and modified the goto labels.